### ISSUE TITLE
There needs to be a test for the GET /assessments route

### ISSUE THREAD

**Issue by Awatanka** [OpenTree-Education/rhizone-lms#536]
### Describe the Feature / Enhancement

There needs to be a test for the GET `/assessments` route (from the router file), which currently does not exist. We need to test this route to ensure the code written for the test will behave as expected by the protocol we previously defined.

This issue continues the work from issue #515.

### Additional Details and Resources

This route has no URL parameters and expects no body in the request. The response should contain the following:

#### Facilitator:
```typescript
collectionEnvelope([{
  curriculum_assessment: CurriculumAssessment;
  program_assessment: ProgramAssessment;
  submissions_summary: FacilitatorAssessmentSubmissionsSummary;
}])
```

#### Participant:
```typescript
collectionEnvelope([{
  curriculum_assessment: CurriculumAssessment;
  program_assessment: ProgramAssessment;
  submissions_summary: AssessmentSubmissionsSummary;
}])
```

### Correct Project Selected

- [X] This issue is linked to the appropriate project

### Labels

- [X] This issue has appropriate labels.

**Comment by Awatanka** [comment]
EOD: worked on the tests for GET /assessments route. I added mock data and implemented several tests for the functions such as `principalEnrolledPrograms`, `getAssessmentsForProgram`, `findRoleInProgram`, and `getProgramIdByCurriculumAssessmentId`. These functions help retrieve information about enrolled programs, assessments for a program, participant roles, and program IDs for a given curriculum assessment. 
Plan for tomorrow: 

- [x] check tests for `principalEnrolledPrograms`, `getAssessmentsForProgram`, `findRoleInProgram`, and `getProgramIdByCurriculumAssessmentId`; 
- [x] write tests for `getAssessmentSubmissionsSummary`, `getProgramIdByProgramAssessmentId`, `getCurriculumAssessment`,   `getFacilitatorAssessmentSubmissionsSummary`

**Comment by Awatanka** [comment]
EOD:  I worked on assessmentsRouter tests for `get/assessments` route. I added a test for the 'unenrolled participant' route behavior. It was successful only in cases of mocking the general function.  I added a draft for future tests focused on the router's behavior when there are different roles like the facilitator, participant, or unenrolled user. I added mocked data and functions for testing the route based on participant role, but tests have failed. I tried to figure out the reason and used different approaches in mocking and writing tests. 

Plan for tomorrow: 
- [x] ask for a code review from @seidior in order to figure out what was the issue in order to be able to continue working as soon as possible to finish the task tomorrow. 
- [x] fixed issue with tests for `get/assessments` - 'participant' role 
- [x] start and finish with tests for `get/assessments` - 'facilitator' role
